packages/docs-page/server.js

```diff
@@ -95,49 +97,78 @@ async function generateStaticProps({
+      if (isLatest) {
+        latestVersion = version
+      }
+      return {
+        name: isLatest ? 'latest' : version,
+        label: isLatest ? `${version} (latest)` : version,
+      }
+    })
 
     // Only load docs content from the DB if we're in production or there's an explicit version in the path
     // Preview and dev environments will read the "latest" content from the filesystem
-    if (process.env.VERCEL_ENV === 'production' || versionFromPath) {
+    if (
+      process.env.VERCEL_ENV === 'production' &&
+      versionFromPath !== 'latest'
+    ) {
       // remove trailing index to ensure we fetch the right document from the DB
-      const paramsNoIndex = (params[paramId] ?? []).filter(
-        (param, idx) =>
-          !(param === 'index' && idx === params[paramId].length - 1)
+      const pathParamsNoIndex = paramsNoVersion.filter(
+        (param, idx, arr) => !(param === 'index' && idx === arr.length - 1)
       )
-      const pagePathToLoad = versionFromPath
-        ? [basePath, ...paramsNoIndex].join('/')
-        : [basePath, currentVersionNormalized, ...paramsNoIndex].join('/')
 
-      let doc
-      const [{ mdxSource }, navData] = await Promise.all([
-        loadVersionedDocument(product.slug, pagePathToLoad).then(
-          (docResult) => {
-            doc = docResult
-            return mdxRenderer(docResult.markdownSource)
-          }
-        ),
-        cachedLoadVersionNavData(
-          product.slug,
-          basePath,
-          versionFromPath ?? currentVersionNormalized
-        ),
+      const currentVersionNormalized = normalizeVersion(versionFromPath)
+
+      const _fullPath = [
+        'doc',
+        currentVersionNormalized,
+        basePath,
+        ...pathParamsNoIndex,
+      ].join('/')
+
+      const documentPromise = fetchDocument(productSlug, _fullPath)
+      const navDataPromise = cachedFetchNavData(
+        productSlug,
+        basePath,
+        currentVersionNormalized
+      )
+
+      const [document, navData] = await Promise.all([
+        documentPromise,
+        navDataPromise,
       ])
 
+      const { mdxSource } = await mdxRenderer(document.markdownSource)
+      const frontMatter = document.metadata
+
       // Construct the githubFileUrl, used for "Edit this page" link
-      const githubFileUrl = `https://github.com/hashicorp/${product.slug}/blob/${doc.gitRef}/website/content/${doc.filePath}`
+      // https://github.com/hashicorp/waypoint/blob/main/website/content/commands/index.mdx
+      // https://github.com/hashicorp/waypoint/blob/e591452/website/content/commands/index.mdx
+      // https://github.com/hashicorp/waypoint/blob/e591452e6f9a155a038d45158e05d904ba006d37/website/content/commands/index.mdx
+      // const githubFileUrl = `https://github.com/hashicorp/${productSlug}/blob/${document.sha}/website/content/${doc.filePath}`
+
+      // UNSUPPORTED?
+      // Must be serializeable
+      const githubFileUrl = null
```

Comment:
This should be achievable once we store the full filepath! ref: https://app.asana.com/0/1100423001970639/1201024261811837/f

Reply:
Left a Todo